hrm-domain/hrm-service/src/contact/contact.ts

```diff
@@ -117,11 +158,16 @@ const permissionsBasedTransformations: PermissionsBasedTransformation[] = [
 
 export const bindApplyTransformations =
   (can: ReturnType<typeof setupCanForRules>, user: TwilioUser) => (contact: Contact) => {
-    return permissionsBasedTransformations.reduce(
+    const permissionsBasedTransformed = permissionsBasedTransformations.reduce(
       (transformed, { action, transformation }) =>
         !can(user, action, contact) ? transformation(transformed) : transformed,
       contact,
     );
+
+    // TODO: Remove once all Flex clients are using new ConversationMedia model
+    const transformed = addLegacyConversationMedia(permissionsBasedTransformed);
+
+    return transformed;
```

Comment:
Emphasize this has to be last step for permission reasons, and make sure we are filtering legacy stuff from blob.